### PR TITLE
[GEOS-9823] [GEOS-9825] Added support for POST GetMap requests adhering to the WMS 1.3.0/SLD 1.1 XML Schema + advertised POST GetMap support in the WMS 1.3.0 capabilities

### PR DESCRIPTION
**Pull request by robinhoutmeyers**
[GEOS-9823](https://osgeo-org.atlassian.net/browse/GEOS-9823)

These commits have 2 goals:
* add support for the SLD 1.1 WMS GetMap XML Schema from http://schemas.opengis.net/sld/1.1/GetMap.xsd (GEOS-9823)
* advertise GetMap POST support in the WMS 1.3.0 capabilities, which allows WMS clients to automatically discover POST as one of the supported HTTP request methods (GEOS-9825).  Geoserver did already support GetMap POST for an older WMS GetMap XML Schema; by means of these commits, it will now also support the SLD 1.1 WMS GetMap XML Schema.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**

For all pull requests:

- [ ] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [ ] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [ ] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

**Review by mprins — CHANGES_REQUESTED** [review]
QA run failed with "Found 7 non-complying files" please run the formatter before committing, as described in https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md#commit-guidelines

also a great deal of tests are failing and there are public API changes, such as removing class `GetMapXmlReader`

File: src/wms/src/test/java/org/geoserver/wms/map/GetMapOwsXmlReaderTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package org.geoserver.wms.map;
```
Comment (aaime) [inline src/wms/src/test/java/org/geoserver/wms/map/GetMapOwsXmlReaderTest.java:1]: Lacks copyright header

File: src/wms/src/main/java/org/geoserver/wms/map/GetMapOwsXmlReader.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package org.geoserver.wms.map;
```
Comment (aaime) [inline src/wms/src/main/java/org/geoserver/wms/map/GetMapOwsXmlReader.java:1]: Lacks copyright header

File: src/wms/src/main/java/org/geoserver/wms/map/GetMapSldXmlReader.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
+package org.geoserver.wms.map;
```
Comment (aaime) [inline src/wms/src/main/java/org/geoserver/wms/map/GetMapSldXmlReader.java:1]: Lacks copyright header

File: src/wms/src/test/java/org/geoserver/wms/map/GetMapSldXmlReaderTest.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package org.geoserver.wms.map;
```
Comment (aaime) [inline src/wms/src/test/java/org/geoserver/wms/map/GetMapSldXmlReaderTest.java:1]: Lacks copyright header

File: src/wms/src/test/java/org/geoserver/wms/wms_1_3/CapabilitiesAllowedRequestTypesTest.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+import org.custommonkey.xmlunit.XMLUnit;
+import org.custommonkey.xmlunit.XpathEngine;
+import org.geoserver.wms.WMSTestSupport;
+import org.junit.Assert;
+import org.junit.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+
+public class CapabilitiesAllowedRequestTypesTest extends WMSTestSupport {
```
Comment (aaime) [inline src/wms/src/test/java/org/geoserver/wms/wms_1_3/CapabilitiesAllowedRequestTypesTest.java:13]: Running tests in GeoServer is slow due to the expensive setup, please do not create a new test class for a single test, merge in an existing test with a similar topic instead. This seems like a good candidate:
https://github.com/geoserver/geoserver/blob/772e58a2d3bbba009f3ab980c2171c31f6bf3f9f/src/wms/src/test/java/org/geoserver/wms/wms_1_3/CapabilitiesIntegrationTest.java

File: src/wms/src/main/java/org/geoserver/wms/map/GetMapSldXmlReader.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+import org.geotools.sld.v1_1.SLD;
+import org.locationtech.jts.geom.Coordinate;
+import org.w3c.dom.Node;
+
+/**
+ * Implementation of AbstractGetMapXmlReader that supports the GetMap XML Schema defined by the
+ * "Styled Layer Descriptor profile of the Web Map Service Implementation Specification", OGC document 05-078r4;
+ * see http://schemas.opengis.net/sld/1.1/GetMap.xsd for the XML Schema.
+ */
+public class GetMapSldXmlReader extends AbstractGetMapXmlReader {
```
Comment (aaime) [inline src/wms/src/main/java/org/geoserver/wms/map/GetMapSldXmlReader.java:16]: This seems incorrect. The schema is part of the SLD 1.1 specification, so I'm assuming it's referring to a SLD 1.1 sld:StyledLayerDescriptor. However, the SLD parser used in the base class can only parse SLD 1.0.
The parsing of SLD 1.1 documents is present, but it's a whole different beast, e.g.:
https://github.com/geoserver/geoserver/blob/e701004e988a8d0a9b7d993090c9a96dcf155e12/src/main/src/main/java/org/geoserver/catalog/SLDHandler.java#L196

**Review by aaime — COMMENTED** [review]
Besides the comments about the code, could you also update the documentation (as per our check-list) mentioning an example of the WMS 1.3 GetMap?

Here is a page that talks about the issue:
https://github.com/geoserver/geoserver/blob/master/doc/en/user/source/services/wms/reference.rst

**Comment by aaime** [comment]
Is this PR still alive? Feedback was provided some 20 days ago, it's up to the PR submitter to update the code so that is passes review.

**Comment by robinhoutmeyers** [comment]
Yes, I am still planning to update it based on the received feedback; I haven't found the time yet but I hope to address the listed comments soon.

**Comment by jodygarnett** [comment]
We are looking at PRs for upcoming release candidate March 1st, do you think this can be ready in time?

**Comment by robinhoutmeyers** [comment]
Hi Jody, this will unfortunately be difficult due to timing issues at my side; if ok, I would like to wrap this up during Q2.

**Comment by aaime** [comment]
Wrapping in Q2 sounds fine, I'll set a not to check back this PR in June then.

**Pull request closed without merging**